FWCore/Framework/src/Principal.cc

```diff
@@ -108,10 +108,10 @@ namespace edm {
     }
   }  // namespace
 
-  //0 means unset
-  static std::atomic<Principal::CacheIdentifier_t> s_nextIdentifier{1};
-  static inline Principal::CacheIdentifier_t nextIdentifier() {
-    return s_nextIdentifier.fetch_add(1, std::memory_order_acq_rel);
+  // Value 0 means unset and is used in Principal constructor. First call to fillPrincipal() will get value 1.
```

Comment:
Won't the first call return 2? You initialize the array elements to 1 and then calls `fetch_add` on each call.

Reply:
The `fetch_add` returns the "old value" (before the increment), i.e. `1` on the first call.